ocaml/typing/outcometree.mli

```diff
@@ -121,7 +122,7 @@ and out_alloc_mode =
 
 type out_class_type =
   | Octy_constr of out_ident * out_type list
-  | Octy_arrow of string * out_type * out_class_type
+  | Octy_arrow of arg_label * out_type * out_class_type
```

Comment:
This surprises me -- do we need this change? Though it all looks to work, so maybe it's not worth undoing.

Reply:
We don't really need it. It seemed a little funny to have one arrow store an arg_label and the other a string. But I can revert this easily